### PR TITLE
docs: agent mode / deployment modes

### PR DESCRIPTION
**Pull request by lennessyy**
## Describe the Change

This PR documents deployment mods and procedure to deploy a cluster in agent mode

## Changed Pages

💻 [Deployment Modes](https://deploy-preview-4099--docs-spectrocloud.netlify.app/deployment-modes/)

## Jira Tickets

🎫 [DOC-1398](https://spectrocloud.atlassian.net/browse/DOC-1398)

## Backports

Can this PR be backported?

- [ ] No.

[DOC-1398]: https://spectrocloud.atlassian.net/browse/DOC-1398?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/docs-content/deployment-modes/agent-mode/manage-agent/localui-theming.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+sidebar_label: "Customize Local UI Theme"
+title: "Customize Local UI Theme - Agent mode"
+description: "Instructions for customizing Local UI theme for hosts deployed in agent mode."
+hide_table_of_contents: false
+sidebar_position: 100
+tags: ["edge"]
+---
+
+Palette offers the option to customize the Local UI web interface. You can change the color of the sidebar as well as
```
Comment (karl-cardenas-coding) [inline docs/docs-content/deployment-modes/agent-mode/manage-agent/localui-theming.md:10]: Maybe I'm missing something but outside of an EdgeForge context. How does a host become a localUI host? It's as simple as the host coming up, registering with Palette and accessing port 5080?

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
Really solid start to this. The overview pages are great. I just have a few suggestions and questions.

File: docs/docs-content/deployment-modes/agent-mode/manage-agent/localui-theming.md (hunk 1 above)
Comment (lennessyy) [reply docs/docs-content/deployment-modes/agent-mode/manage-agent/localui-theming.md:10]: Yes. Local UI is enabled by default. Whether you register with Palette or not, as long as you didn't disable it in your user data, it's going to be available at port 5080.

File: docs/docs-content/deployment-modes/agent-mode/agent-mode.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+tags: ["edge"]
+---
+
+In agent mode, you bring your own host, which can be any host that meets the minimum hardware requirements, regardless
+of the environment. This includes options like an AWS EC2 instance or a Raspberry Pi in your own home. All you need to
+do to have Palette manage your host is to [download and install the Palette agent](./install-agent-host.md).
+
+The diagram below illustrates the agent mode cluster provisioning workflow.
+
+![Architecture Diagram for Agent Mode](/deployment-modes_agent-mode.webp)
```
[the remark this answers is not in the document]
Comment (lennessyy) [reply docs/docs-content/deployment-modes/agent-mode/agent-mode.md:20]: I don't know if adding the sequence directly in the diagram will be helpful in this case. Some of the steps are not sequential - the diagram covers two workflows between Palette management plane and Local UI. I can add a list to summarize the order of operations below the diagram.

[the remark this answers is not in the document]
Comment (karl-cardenas-coding) [reply docs/docs-content/deployment-modes/agent-mode/agent-mode.md:20]: This is great @lennessyy

**Review by karl-cardenas-coding — APPROVED** [review]

File: docs/docs-content/deployment-modes/agent-mode/install-agent-host.md [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+
+- Ensure that the host has `Bash` configured as the default shell.
+
+- Ensure the following software is installed and available:
+  - [jq](https://jqlang.github.io/jq/download/)
+  - [Zstandard](https://facebook.github.io/zstd/)
+  - [Rsync](https://github.com/RsyncProject/rsync)
+  - [conntrack](https://conntrack-tools.netfilter.org/downloads.html). This requirement is specific for clusters that
+    use PXKE as the Kubernetes layer.
+  - (Airgap only) [Docker](https://www.docker.com/) is installed and available.
```
Comment (nianyush) [inline docs/docs-content/deployment-modes/agent-mode/install-agent-host.md:63]: why do we need docker for airgap? @lennessyy

**Pull request merged**